ReactiveCocoa/Swift/Errors.swift

```diff
@@ -8,27 +8,16 @@
 
 import Foundation
 
-/// Represents an error that can be sent upon or received from a signal.
-public protocol ErrorType {
-	/// An NSError corresponding to the receiver.
-	var nsError: NSError { get }
-}
-
-extension NSError: ErrorType {
-	public var nsError: NSError {
-		return self
-	}
-}
+/// Workaround for constraining generic type extensions with non-protocol types.
+/// Unfortunately there's no way to guarantee that instances of types conforming
+/// to this protocol can't be created.
+///
+/// See http://www.openradar.me/21512469
+public protocol NoErrorType: ErrorType {}
```

Comment:
If you really want to have a type that can't be created, just say

```swift
public enum NoError: ErrorType {}
```

With no variants, it has literally no inhabiting values.